x-pack/plugins/canvas/canvas_plugin_src/functions/common/axisConfig.ts

```diff
@@ -40,23 +61,27 @@ export const axisConfig = () => ({
       help: 'Increment size between each tick. Use for number axes only',
     },
   },
-  fn: (context, args) => {
-    const positions = ['top', 'bottom', 'left', 'right', ''];
-    if (!positions.includes(args.position)) {
+  fn: (_context, args) => {
+    const { position, min, max, ...rest } = args;
+
+    if (!VALID_POSITIONS.includes(position)) {
       throw new Error(`Invalid position: '${args.position}'`);
     }
 
-    const min = typeof args.min === 'string' ? moment.utc(args.min).valueOf() : args.min;
-    const max = typeof args.max === 'string' ? moment.utc(args.max).valueOf() : args.max;
+    const minVal = typeof min === 'string' ? moment.utc(min).valueOf() : min;
+    const maxVal = typeof max === 'string' ? moment.utc(max).valueOf() : max;
 
-    if (min != null && isNaN(min)) {
+    // This != check is not !== in order to handle NaN cases properly.
```

Comment:
`minVal != null` means not `null` or `undefined` where `minVal !== null` only means not `null`. What does this comparison have to do with `NaN`?

Reply:
![Screen Shot 2019-04-24 at 5 16 07 PM](https://user-images.githubusercontent.com/297604/56697590-ae8af700-66b4-11e9-98bf-59484c7b9e21.png)

Reply:
Ah, I see. You're trying to describe the early failure as handling NaN cases. The comment is pretty confusing... something like "only check isNaN if there is a value" would have made more sense to me.

Reply:
I'll update it.